src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -385,6 +362,10 @@ private void processOffers() {
 
     @Override
     public synchronized void resourceOffers(SchedulerDriver driver, List<Offer> offers) {
+        if (multiThreaded && !processing()) {
+            startProcessing();
+        }
+
         for (Protos.Offer offer : offers) {
             boolean queued = offerQueue.offer(offer);
             if (!queued) {
```

Comment:
do we still need this piece of code? In both cases, we have no-op.

Reply:
good point, we should continue there

Reply:
actually `else` is better